### PR TITLE
[SHELL32] CFSFolder: Add fallback for failing shell extension

### PR DESCRIPTION
**Pull request by learn-more**
JIRA issue: [CORE-17673](https://jira.reactos.org/browse/CORE-17673)

Also sneak the fonts folder icon back in.

### CODE REVIEW

File: dll/win32/shell32/folders/CFSFolder.cpp [hunk 1]
```diff
@@ -883,10 +883,26 @@ HRESULT WINAPI CFSFolder::BindToObject(
     CLSID clsidFolder;
     if (_ILIsFolder(pidl))
     {
-        clsidFolder = CLSID_ShellFSFolder;
-
         if ((pData->uFileAttribs & (FILE_ATTRIBUTE_SYSTEM | FILE_ATTRIBUTE_READONLY)) != 0)
-            SHELL32_GetCLSIDForDirectory(pfti.szTargetParsingName, L"CLSID", &clsidFolder);
+        {
+            hr = SHELL32_GetCLSIDForDirectory(pfti.szTargetParsingName, L"CLSID", &clsidFolder);
+
```
Comment (HBelusca) [inline dll/win32/shell32/folders/CFSFolder.cpp:889]:
```suggestion
```

File: dll/shellext/fontext/fontext.cpp [hunk 2]
```diff
@@ -54,7 +54,9 @@ STDAPI DllGetClassObject(REFCLSID rclsid, REFIID riid, LPVOID *ppv)
 STDAPI DllRegisterServer()
 {
     WCHAR Path[MAX_PATH] = { 0 };
-    static const char DesktopIniContents[] = "[.ShellClassInfo]\r\nCLSID={BD84B380-8CA2-1069-AB1D-08000948F534}\r\n";
+    static const char DesktopIniContents[] = "[.ShellClassInfo]\r\n"
+        "CLSID={BD84B380-8CA2-1069-AB1D-08000948F534}\r\n"
+        "IconResource=%SystemRoot%\\system32\\shell32.dll,-39\r\n"; // IDI_SHELL_FONTS_FOLDER
```
Comment (HBelusca) [inline dll/shellext/fontext/fontext.cpp:59]: try the following:
```suggestion
        "IconResource=%SystemRoot%\\system32\\shell32.dll,-" _CRT_STRINGIZE(IDI_SHELL_FONTS_FOLDER) "\r\n";
```
The `_CRT_STRINGIZE` macro is automatically included from crtdefs.h included as part of the CRT.

Comment (learn-more) [reply dll/shellext/fontext/fontext.cpp:59]: I can't really include shell32's private header that defines `IDI_SHELL_FONTS_FOLDER`, can I?

Comment (HBelusca) [reply dll/shellext/fontext/fontext.cpp:59]: ah yes, my bad (you would need to include shell's own resource.h or whatever name) 😭

**Review by ThFabba — APPROVED** [review]

**Pull request merged**